### PR TITLE
allow creation of an Identity and subsequent use in requests/subscriptions

### PR DESCRIPTION
**Pull request by jonfreedman**
fixes https://github.com/assylias/jBloomberg/issues/40

### CODE REVIEW

**Comment by jonfreedman** [comment]
Needs a further change to allow the creation of a user identity, e.g.

authReq.set("authId", authID);
authReq.set("ipAddress", appIP);

File: src/main/java/com/assylias/jbloomberg/DefaultBloombergSession.java [hunk 1]
```diff
@@ -224,6 +220,65 @@ public synchronized void stop() {
...
+        return getIdentity(authRequest -> {
+            CorrelationID cId = getNextCorrelationId();
+            TokenResultParser tokenResultParser = new TokenResultParser();
+            eventHandler.setParser(cId, tokenResultParser);
+            session.generateToken(cId);
+            TokenResultParser.Result token = tokenResultParser.getResult();
+            if (token.isEmpty()) {
+                throw new IllegalStateException("Failed to generate token - " + token.getError());
+            }
+            authRequest.set("token", tokenResultParser.getRequestResult().getToken());
```
Comment (jonfreedman) [inline src/main/java/com/assylias/jbloomberg/DefaultBloombergSession.java]: tokenResultParser.getRequestResult() should be replaced by token

**Comment by assylias** [comment]
Hi Jon, I've finally managed to review this PR. Unless you have a good reason to oppose, I would prefer to overload the `submit` and `subscribe` methods to accept an `Identity` as an argument instead of adding a `withIdentity` method in every Builder class.

If you are happy with that, I'll push my changes tomorrow.

**Comment by jonfreedman** [comment]
Sounds good to me, thanks for looking at this!

**Comment by assylias** [comment]
As discussed, solved with https://github.com/assylias/jBloomberg/pull/48

**Pull request closed without merging**